client/state/media/reducer.js

```diff
@@ -2,40 +2,79 @@
  * External dependencies
  */
 import { combineReducers } from 'redux';
-import { keyBy, omit } from 'lodash';
+import { omit } from 'lodash';
 
 /**
  * Internal dependencies
  */
-import { MEDIA_DELETE, MEDIA_RECEIVE } from 'state/action-types';
+import {
+	MEDIA_DELETE,
+	MEDIA_RECEIVE,
+	MEDIA_REQUEST_FAILURE,
+	MEDIA_REQUESTING } from 'state/action-types';
 import { createReducer } from 'state/utils';
+import MediaQueryManager from 'lib/query-manager/media';
 
-export const items = createReducer( {}, {
-	[ MEDIA_RECEIVE ]: ( state, action ) => {
-		const { siteId, media } = action;
+export const queries = ( () => {
+	function applyToManager( state, siteId, method, createDefault, ...args ) {
+		if ( ! state[ siteId ] ) {
+			if ( ! createDefault ) {
+				return state;
+			}
+
+			return {
+				...state,
+				[ siteId ]: ( new MediaQueryManager() )[ method ]( ...args )
+			};
+		}
+
+		const nextManager = state[ siteId ][ method ]( ...args );
+
+		if ( nextManager === state[ siteId ] ) {
+			return state;
+		}
+
+		return {
+			...state,
+			[ siteId ]: nextManager
+		};
+	}
```

Comment:
I copied applyToManager from state/posts :) Not sure what's best here. Looks like it could be abstracted. See https://github.com/Automattic/wp-calypso/pull/10392#issuecomment-270269481.